src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java

```diff
@@ -361,13 +362,12 @@ public Map<TopicPartition, OffsetAndMetadata> preCommit(
       offsets.forEach(
           (topicPartition, offsetAndMetadata) -> {
             long offSet = sink.getOffset(topicPartition);
-            if (offSet != 0) {
+            if (offSet != NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE) {
               committedOffsets.put(topicPartition, new OffsetAndMetadata(offSet));
             }
           });
     } catch (Exception e) {
       this.DYNAMIC_LOGGER.error("PreCommit error: {} ", e.getMessage());
-      return new HashMap<>();
```

Comment:
This is a new change, i think we should still return the offsets for the partitions that don't have an issue

Reply:
i think we should still return the offsets for the partitions that don't have an issue
> 
But you are returning empty here. Am I missing?

Reply:
ahh sorry.